source/socketstuff.lua

```diff
@@ -11,183 +11,192 @@ gbolIsAHost = false                					-- defaults to NOT a host until the play
 Put the following code into love.update so that the host can do host things:
 
 	if gbolIsAHost then
-		ss.HostListenPort()
-
+		ss.hostListenPort()
+		
 		-- get just one item from the queue and process it
-		local incoming = ss.GetItemInHostQueue()		-- could be nil
+		local incoming = ss.getItemInHostQueue()		-- could be nil
 		if incoming ~= nil then
 			print(inspect(incoming))
 		end
-
+	
 		msg = whatever		-- string, number or table.
-		ss.AddItemToHostOutgoingQueue(msg)
-		ss.SendToClients()
+		ss.addItemToHostOutgoingQueue(msg)
+		ss.sendToClients()
 		msg = {}
 	end
 
 Put the following code into love.update so that clients can do client things:
 
 	if gbolIsAClient then
-		ss.ClientListenPort()
-
+		ss.clientListenPort()
+		
 		-- get just one item from the queue and process it
-		local incoming = ss.GetItemInClientQueue()		-- could be nil
+		local incoming = ss.getItemInClientQueue()		-- could be nil
 		if incoming ~= nil then
 			print(inspect(msg))
 		end
 
 		msg = whatever 		-- string, number or table.
-		ss.AddItemToClientOutgoingQueue(msg)	--
-		ss.SendToHost()
+		ss.addItemToClientOutgoingQueue(msg)	-- 
+		ss.sendToHost()
 		msg = {}
 	end
-
+	
 ]]
 
 
 
 local socketstuff = {}
 
-local arrHostIncomingQueue = {}
-local arrClientIncomingQueue = {}
-local arrHostOutgoingQueue = {}
-local arrClientOutgoingQueue = {}
-local arrClientNodes = {}
+local hostIncomingQueue = {}
+local clientIncomingQueue = {}
+local hostOutgoingQueue = {}
+local clientOutgoingQueue = {}
+local cilentNodes = {}
 
-udpclient = nil
-udphost = nil
+local udpClient = nil
+local udpHost = nil
 
-function socketstuff.HostListenPort()
+function socketstuff.hostListenPort()
 -- listens for a message and adds it to the queue
-    local data, ip, port = udphost:receivefrom()
+    local data, ip, port = udpHost:receivefrom()
 	local unpackeddata
     if data then
 		unpackeddata = bitser.loads(data)
-        table.insert(arrHostIncomingQueue,unpackeddata)
+        table.insert(hostIncomingQueue,unpackeddata)
     end
-    --socket.sleep(0.01)    -- this doesn't seem to do much so I removed it
-
+    -- socket.sleep(0.01)    -- this doesn't seem to do much so I removed it
+	
 	local node = {}
     node.ip = ip
     node.port = port
-
+	
 	if port == nil or unpackeddata == nil then
 		-- no message, do nothing
 	else
 		local bolAddClient = true
-		for k,v in pairs(arrClientNodes) do
+		for k,v in pairs(cilentNodes) do
 			if node.ip == v.ip and node.port == v.port then
 				-- this node is already captured
 				bolAddClient = false

```

Comment:
bol shouldn't be there anymore. Also, we might be able to get rid of the boolean entirely by doing:
```
		for k,v in pairs(cilentNodes) do
			if node.ip ~= v.ip and node.port ~= v.port then
				-- new client node
				table.insert(cilentNodes,node)
				break
			end
		end
```
Without the extra if statement. (untested).
Also, I'm unsure what the code exactly does, but I feel like the port should be the same for all clients, otherwise they wouldn't be able to communicate with each other as far as I know. So possible it would be enough to do:
 `if node.ip ~= v.ip then`
 But I've never worked on any multiplayer / networking code, so maybe I'm wrong.

Reply:
I have been taught that 'break'is bad style and should be avoided by exciting a loop via booleans. We should get a consensus and put in style guide.

@MadByteDE  @Philbywhizz

Reply:
The server has a port all clients need to use but each client has a different ip and port and the host needs to track that.

Reply:
I don't consider a 'break' to be bad style, as long as it is clear as to what is being 'broken' out of. It's part of the language, so use it.

I've seen people try and avoid using break using a boolean or other variable in the condition statements, and can be convoluted and hard to understand to do it that way.

Clean and easy readable code should always win. :)

Reply:
Glad you said that. I use break in other places and is my preference.  :)